walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,210 @@
+                attandeeSchedule.add(timeRange);
+                optionalSchedules.put(attandee, attandeeSchedule);
+                allSchedules.put("optional", optionalSchedules);
+            }
+            
+        }
+    }
+    return allSchedules;
+  }
+  /*Given two overlapped TimeRanges, return the merged TimeRange
+  param: @one and @two: overlapped TimeRanges
+  return: @mergeTimeSlot: merge TimeRange
+  */
+  public static TimeRange mergeTwoOverlappedTimeRanges(TimeRange one, TimeRange two) {
+    int start = (one.start() < two.start())? one.start() : two.start();
+    int end = (one.end() > two.end())? one.end() : two.end();
+    TimeRange mergedTimeSlot = TimeRange.fromStartEnd(start, end, false);
+    return mergedTimeSlot;
+  }
+
+  /*Given a List of acummulated TimeRange and a TimeRange to add (assuming the TimeRange can overlap with the last TimeRange in the list)
+  param: @accum: the list of accumuated TimeRange
+         @newTimeRange: to be added
+  return @accum the updated accumulated list
+  */
+  public static ArrayList<TimeRange> mergeTimeRangeWithAccumulatedTimeRanges(ArrayList<TimeRange> accum, TimeRange newTimeRange) {
+    TimeRange lastTimeRange = (accum.size() == 0)? null: accum.get(accum.size() - 1);
+    if (lastTimeRange == null) {
+        ArrayList<TimeRange> newAccum = new ArrayList<>();
+        newAccum.add(newTimeRange);
+        return newAccum;
+    }
+    else {
+        if (lastTimeRange.overlaps(newTimeRange)) {
+            accum.remove(accum.size() - 1);
+            TimeRange mergedTimeRange = mergeTwoOverlappedTimeRanges(lastTimeRange, newTimeRange);
+            accum.add(mergedTimeRange);
+        }
+        else {
+            accum.add(newTimeRange);
+        }
+        return accum;
+    }    
+  }
+
+  /* Given a list of schedules and an optional list of pre-existing TimeRange, return a new list of TimeRange that merges all the schedules and pre-existing TimeRange
+  param: @allSchedules: a hashMap whose key is attandee, value is the schedule of that attandee
+         @preExistingTimeRanges: a list of pre-existing TimeRange or null
+  return: @possibleSlots: a list of possible TimeRange to fit all mandatory attandees
+  */
+  public static List<TimeRange> mergeAllSchedules(HashMap<String, ArrayList<TimeRange>> allSchedules, List<TimeRange> preExistingTimeRanges) {
+    // Merge all schedules of attandees using merging two sorted list algorithm
+    List<TimeRange> mergeSchedule = (preExistingTimeRanges != null)? preExistingTimeRanges: new ArrayList<>();
+
+    for (String attandee: allSchedules.keySet()) {
+        ArrayList<TimeRange> attandeeSchedule = allSchedules.get(attandee);
+        Comparator<TimeRange> comparatorByStart = TimeRange.ORDER_BY_START;
+        Collections.sort(attandeeSchedule, comparatorByStart);
+        allSchedules.put(attandee, attandeeSchedule);
+        
+        System.out.println(attandeeSchedule); 
```

Comment:
Nit: You probably put this here for debugging, should be removed in the final version.